File: asion-account/asion-account-common/src/main/java/org/asion/account/domain/application/AccountManager.java
package org.asion.account.domain.application;

import org.asion.account.domain.model.Account;

/**
 * @author Asion.
 * @since 16/5/29.
 */
public interface AccountManager {
    Account save(Account account);
    Account findOne(Long id);
    void delete(Long id);
    Iterable<Account> findAll();

    Account login(Account account);
}

File: asion-account/asion-account-common/src/main/java/io/vincent/account/domain/application/AccountManager.kt
package io.vincent.account.domain.application

import io.vincent.account.domain.model.Account
import io.vincent.common.vo.Page
import org.asion.base.ddd.application.RemoteFacade
import java.util.*

/**
 * 账号基础接口。
 *
 * @author Asion.
 * @since 16/5/29.
 */
@RemoteFacade
interface AccountManager {
    fun create(account: Account): Account
    fun update(account: Account): Account
    fun findOne(id: Long): Account?
    fun findById(id: Long): Optional<Account>
    fun findPage(pageNumber: Int, pageSize: Int): Page<Account>
}
